pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -26,23 +27,40 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ecr"
 	"github.com/golang/glog"
-	"k8s.io/kubernetes/pkg/cloudprovider"
 	"k8s.io/kubernetes/pkg/credentialprovider"
 )
 
-var registryUrls = []string{"*.dkr.ecr.*.amazonaws.com"}
+// AWSRegions is the complete list of regions known to the AWS cloudprovider
+// and credentialprovider.
+var AWSRegions = [...]string{
+	"us-east-1",
+	"us-west-1",
+	"us-west-2",
+	"eu-west-1",
+	"eu-central-1",
+	"ap-southeast-1",
+	"ap-southeast-2",
+	"ap-northeast-1",
+	"ap-northeast-2",
+	"cn-north-1",
+	"us-gov-west-1",
+	"sa-east-1",
+}
+
+const registryURLTemplate = "*.dkr.ecr.%s.amazonaws.com"
 
 // awsHandlerLogger is a handler that logs all AWS SDK requests
 // Copied from cloudprovider/aws/log_handler.go
 func awsHandlerLogger(req *request.Request) {
 	service := req.ClientInfo.ServiceName
+	region := req.Config.Region
 
 	name := "?"
 	if req.Operation != nil {
 		name = req.Operation.Name
 	}
 
-	glog.V(4).Infof("AWS request: %s %s", service, name)
+	glog.V(3).Infof("AWS request: %s:%s in %s", service, name, *region)
```

Comment:
I think we probably should guard against the region being nil, but there's only a tiny number of non-regional calls AFAIK, and I think they are all S3 calls.